daemon/monitor.go

```diff
@@ -32,6 +32,16 @@ func (daemon *Daemon) handleContainerExit(c *container.Container, e *libcontaine
 	var ctrExitStatus container.ExitStatus
 	c.Lock()
 
+	// If the container is not running, paused, or restarting, then the exit
+	// event was triggered while the container was being created. This
+	// indicates that the networking config, and thus the container spec, is
+	// temporarily or permanently invalid. In that case, we don't want to
+	// restart the container.
+	if !c.State.Running && !c.State.Paused && !c.State.Restarting {
```

Comment:
@vvoland Could you confirm whether this change is reasonable? As mentioned in the commit message / PR description, this is also going to ignore any error happening during the following call to c8d. I'm not 100% sure if this is okay, or if we do want to trigger the restart process if this call fails.

https://github.com/moby/moby/blob/af898abe44662d9554fb15ee4d4a7307f1b8e315/daemon/start.go#L240-L242

Reply:
Shouldn't we still execute the rest of the function though? The task was still created, so the cleanup should still happen I think?

Reply:
We should only make sure that we don't restart the container. Perhaps we could also make use of `c.State.Err() != nil`.

Reply:
> Shouldn't we still execute the rest of the function though? The task was still created, so the cleanup should still happen I think?

As dicussed on Slack, we need to early return from handleContainerExit to keep the container in Created state (as it was before 28.0). Otherwise the container will be reported in Exited state with exit code == 137.